### ISSUE TITLE
【FR】Bug: operator accuracy degradation on nvidia (baidu/ERNIE-4.5-21B-A3B-PT)

### ISSUE THREAD

**Issue by Lxiparer** [FlagOpen/FlagGems#4554]
### Description

On ERNIE-4.5-21B-A3B-PT, **10 specific operators were identified** as causing accuracy degradation. After disabling them, accuracy recovered from 84% to 90% (matching native). The remaining concern is **borderline performance at 84.4%** after those 10 operators are disabled.

### Environment

| Item | Value |
|------|-------|
| Hardware | nvidia |
| GPU | H20-3e |
| PyTorch | 2.11.0+cu130 |
| FlagGems | 5.0.1.rc.0 |
| vLLM | 0.20.2 |

### Affected Operators (confirmed)

Accuracy tuning (`traces/05_accuracy_tuning.json`) successfully isolated the problematic operators in 2 rounds:

**diagnose_ops.py accuracy-groups**: 5 operator groups identified — `memory(3)`, `math(7)`, `index(2)`, `reduce(4)`, `other(14)`

| Round | Group | Disabled Operators | Score | Drop | Result |
|-------|-------|--------------------|-------|------|--------|
| 1 | memory | `zeros, ones, full` | 84.0% | 6.0% | FAILED |
| 2 | memory + math | `zeros, ones, full, cos, sin, reciprocal, add, sub, mul, div` | 90.0% | 0.0% | **PASSED** |

**10 problematic operators identified** (from `results/accuracy_tuned_control.json`):
- **memory group**: `zeros, ones, full`
- **math group**: `cos, sin, reciprocal, add, sub, mul, div`

**20 operators kept enabled**: `arange, argmax, cat, copy, cumsum, exponential_, fill, flip, index, le, lt, masked_fill, pow, rand, rand_like, scatter, softmax, sort, to, where`

### Remaining Issue: Borderline Performance

After disabling the 10 accuracy-offending operators, **20 operators remain active**. The performance with these 20 operators:

| Mode | Native | FlagGems | Ratio |
|------|--------|----------|-------|
| Standard | 12,406 tok/s | 10,466 tok/s | **84.4%** (borderline, threshold 80%) |
| Plugin | 12,406 tok/s | 9,751 tok/s | **78.6%** (below threshold) |

The `general_mm` operator (hopper, shape [-, 8192, 3584, 2560]) is active and a prime suspect for the 15.6% throughput gap due to its large matrix dimensions.

### Steps to Reproduce

1. Environment: nvidia H20-3e, PyTorch 2.11.0+cu130, FlagGems 5.0.1.rc.0, vLLM 0.20.2
2. Enable FlagGems with 30 initial operators
3. Run GPQA evaluation — observe V1=90% vs V2=84% (6% drop)
4. Run `diagnose_ops.py accuracy-groups` → identifies 5 groups
5. Disable `memory` group → 84% (still 6% drop)
6. Disable `memory` + `math` group → 90% (0% drop, recovered)
7. The 10 identified operators are now disabled; investigate which of the remaining 20 cause the 15.6% performance gap

### Actual Behavior

```
Accuracy:
  V1 (native)            : 90.0%
  V2 (full FlagGems, 30) : 84.0%  ← 6% drop
  V2 (disable memory)    : 84.0%  ← still 6% drop
  V2 (disable mem+math)  : 90.0%  ← recovered to native

Performance (with 10 disabled, 20 active):
  Standard               : 84.4%  ← borderline
  Plugin                 : 78.6%  ← below 80% threshold
```

### FlagGems Integration Code

**Current gems.txt** (30 operators from initial run): zeros, full, zero_, arange, div, pow, reciprocal, mul, cos, sin, cat, to_copy, ones, fill_scalar_, flip, general_mm (hopper, [-,8192,3584,2560]), copy_, index, rand_like, argmax, lt_scalar, where_self, where_self_out, true_divide_, softmax, exponential_, rand, sub, add, scatter_

### Possible Directions

**Accuracy (resolved)**: The 10 math/memory operators (`zeros, ones, full, cos, sin, reciprocal, add, sub, mul, div`) have precision issues that should be investigated at the kernel level, even though they are now disabled for this model.

**Performance (remaining)**:
- Profile `general_mm` with shape [batch, 8192, 3584, 2560] — the largest matmul in this model — to find the 15.6% throughput gap
- Run operator elimination search (performance mode) on the remaining 20 operators to quantify each operator's throughput impact
- Plugin mode at 78.6% introduces additional dispatch overhead (`vllm_fl.dispatch.ops.*`) that warrants separate investigation
